src/core/auth/qgsauthmethod.h

```diff
@@ -173,7 +178,6 @@ class CORE_EXPORT QgsAuthMethod : public QObject
      * \note Non-public since this is an abstract base class
      */
     explicit QgsAuthMethod()
-      : mMutex( QMutex::RecursionMode::Recursive )
```

Comment:
#if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
    : mMutex( QMutex::RecursionMode::Recursive )
#endif

(github won't let me suggest this!)